lib/player.rb

```diff
@@ -0,0 +1,36 @@
+class Player
+  attr_reader :score, :inputs, :mark
+
+  def initialize (mark)
+    @mark = mark
+    @score = 0
+    @inputs = []
+  end
+
+  def save (input)
+    @inputs << input
+  end
+
+  def valid? (input)
+    if 1 <= input && input <= 9
+      @inputs.each { |x| return false if input == x }
+    else
+      return false
+    end
+    true
+  end
+
+  def victory?
```

Comment:
This method of finding out the winner or if a player has already won should be the responsibility of the Game class. The player's sole responsibility is making moves (playing the game).